ducktape/tests/test.py

```diff
@@ -173,20 +173,79 @@ def _escape_pathname(s):
     return re.sub("^\.|\.$", "", s)
 
 
-class TestContext(Logger):
+def test_logger(logger_name, log_dir, debug, max_parallel):
+    """Helper method for getting a test logger object
+
+    Note that if this method is called multiple times with the same logger_name, it returns the same logger object.
+    Note also, that for a fixed logger_name, configuration occurs only the first time this function is called.
+    """
+    return TestLogger(logger_name, log_dir, debug, max_parallel).logger
+
+
+class TestLogger(Logger):
+    def __init__(self, logger_name, log_dir, debug, max_parallel):
+        self.logger_name = logger_name
+        self.log_dir = log_dir
+        self.debug = debug
+        self.max_parallel = max_parallel
+
+    def configure_logger(self):
+        """Set up the logger to log to stdout and files.
+        This creates a directory and a few files as a side-effect.
+        """
+        if self.configured:
+            return
+
+        self._logger.setLevel(logging.DEBUG)
+        mkdir_p(self.log_dir)
+
+        # Create info and debug level handlers to pipe to log files
+        info_fh = logging.FileHandler(os.path.join(self.log_dir, "test_log.info"))
+        debug_fh = logging.FileHandler(os.path.join(self.log_dir, "test_log.debug"))
+
+        info_fh.setLevel(logging.INFO)
+        debug_fh.setLevel(logging.DEBUG)
+
+        formatter = logging.Formatter(ConsoleDefaults.TEST_LOG_FORMATTER)
+        info_fh.setFormatter(formatter)
+        debug_fh.setFormatter(formatter)
+
+        self._logger.addHandler(info_fh)
+        self._logger.addHandler(debug_fh)
+
+        ch = logging.StreamHandler(sys.stdout)
+        ch.setFormatter(formatter)
+        if self.debug and self.max_parallel == 1:
```

Comment:
Still concerned about why we're doing this. If the user asks for debug, seems like they should get debug level output... `max_parallel` shouldn't have any impact on their request.